### DISCUSSION TITLE
Radio Settings

### DISCUSSION THREAD

**Discussion by Kayto** [Kayto/Armachat-circuitpython#3] [General]
Currently config.py settings are not parsed i.e. spread, bandwidth and rate? The library does not support this only groups.

Frequency settings are fixed in main code.py and use the ulora.py library.

LINE 612
    _rfm9x = ulora.LoRa(
        spi, CS, modem_config=ulora.ModemConfig.**Bw125Cr45Sf2048**, tx_power=config.power_

https://meshtastic.org/docs/developers/device/radio-settings

1. Easier to use recommended configuration groups instead of separate config spread/bitrate .... these set groups need to be implemented in the code rather than individual spread/bitrates. These need to be selectable in the config.py and parsed to the code.py

                  1.  Bw125Cr45Sf128 = Default medium range
                  2.  Bw500Cr45Sf128 = Fast+short range
                  3.  Bw31_25Cr48Sf512 = Slow+long range
                  4.  Bw125Cr48Sf4096 =  Slow+long range
                  5.  Bw125Cr45Sf2048 =  Slow+long range
                  6.  Bw31Cr48Sf4096 =  Slow+Extra long range

2.  The above need checking .Additional configuration combination to add: 0.046 kbps12 / 40964/831160dBTwice the range and/or coverage of "Long Slow", high resilience to noise

3. need to check that the code.py radio setting overides the ulora.py

**Comment by Kayto** [comment]
Points 1, 2 and 3 now implemented. The above profiles has now been implemented in the code by Bobricius. Great work.
There is an issue that the for the new profile to take effect it looks like a reboot is required.

4. reloading profiles needs implementation - either reboot or reset up of radio upon profile change.

**Reply by richteel** [reply]
This should be working now. I tested the changes and the appear to work. I changed one Armachat and sent a message. It was not received on the other one. I then changed the setting on the other Armachat and sent a test message and it was received.

**Comment by Kayto** [comment]
I have not tested this fully. I had an issue on my two having different radio settings on start up which caused a problem as you mention above.

It does raise another item on being able to send a "scanning ping" on all settings - to be able to notify the reciever of which setting channel you are on! I might look at this as part of the ping messaging item in my wip_code.py.

**Reply by richteel** [reply]
Interesting idea
